--- buildroot/package/base-files/image-config.in.orig	2019-09-07 16:01:13.000000000 +0200
+++ buildroot/package/base-files/image-config.in	2019-09-07 16:12:47.082987559 +0200
@@ -183,7 +183,7 @@
 	config VERSION_REPO
 		string
 		prompt "Release repository"
-		default "http://downloads.openwrt.org/releases/18.06-SNAPSHOT"
+		default "https://github.com/freifunk-dresden/firmware-freifunk-dresden.git"
 		help
 			This is the repository address embedded in the image, it defaults
 			to the trunk snapshot repo; the url may contain the following placeholders:
@@ -210,6 +210,7 @@

 	config VERSION_MANUFACTURER
 		string
+		default "Freifunk Dresden"
 		prompt "Manufacturer name"
 		help
 			This is the manufacturer name embedded in /etc/device_info
@@ -251,7 +252,7 @@
 	config VERSION_FILENAMES
 		bool
 		prompt "Version number in filenames"
-		default y
+		default n
 		help
 			Enable this to include the version number in firmware image, SDK-
 			and Image Builder archive file names
@@ -259,7 +260,7 @@
 	config VERSION_CODE_FILENAMES
 		bool
 		prompt "Revision code in filenames"
-		default y
+		default n
 		help
 			Enable this to include the revision identifier or the configured
 			version code into the firmware image, SDK- and Image Builder archive
