--- buildroot/package/kernel/mac80211/files/lib/wifi/mac80211.sh	2018-04-01 16:49:08.940983457 +0200
+++ buildroot/package/kernel/mac80211/files/lib/wifi/mac80211.sh	2018-04-01 16:59:21.220767808 +0200
@@ -125,7 +125,9 @@
 			set wireless.default_radio${devidx}.ssid=LEDE
 			set wireless.default_radio${devidx}.encryption=none
 EOF
-		uci -q commit wireless
+		# always create config in /var/etc/config and symlink from /etc/config
+		# this allows me to change wifi without writing to flash
+		uci -q -c /var/etc/config commit wireless

 		devidx=$(($devidx + 1))
 	done
