--- sources/Makefile	2013-10-16 00:28:45.761116098 +0200
+++ sources/Makefile-neu	2013-10-16 01:07:51.519079729 +0200
@@ -34,19 +34,19 @@
 # CFLAGS += -DAVL_DEBUG -DAVL_TEST
 
 # optional defines (you may disable these features if you dont need them)
-# CFLAGS += -DNO_DEBUG_ALL
-# CFLAGS += -DNO_DEBUG_DUMP
-# CFLAGS += -DNO_DEBUG_TRACK
-# CFLAGS += -DNO_DEBUG_SYS
+CFLAGS += -DNO_DEBUG_ALL
+CFLAGS += -DNO_DEBUG_DUMP
+CFLAGS += -DNO_DEBUG_TRACK
+CFLAGS += -DNO_DEBUG_SYS
 # CFLAGS += -DLESS_OPTIONS
-# CFLAGS += -DNO_TRAFFIC_DUMP
-# CFLAGS += -DNO_DYN_PLUGIN
-# CFLAGS += -DNO_TRACE_FUNCTION_CALLS
-# CFLAGS += -DNO_DEBUG_MALLOC
-# CFLAGS += -DNO_MEMORY_USAGE
+CFLAGS += -DNO_TRAFFIC_DUMP
+CFLAGS += -DNO_DYN_PLUGIN
+CFLAGS += -DNO_TRACE_FUNCTION_CALLS
+CFLAGS += -DNO_DEBUG_MALLOC
+CFLAGS += -DNO_MEMORY_USAGE
 
 # experimental or advanced defines (please dont touch):
-# CFLAGS += -DNO_ASSERTIONS       # (disable syntax error checking and error-code creation!)
+CFLAGS += -DNO_ASSERTIONS       # (disable syntax error checking and error-code creation!)
 # CFLAGS += -DEXTREME_PARANOIA    # (check difficult syntax errors)
 # CFLAGS += -DEXIT_ON_ERROR       # (exit and return code due to unusual behavior)
 # CFLAGS += -DTEST_DEBUG
