diff -Naur --suppress-blank-empty --strip-trailing-cr buildroot/target/linux/ar71xx/config-3.3 buildroot.c/target/linux/ar71xx/config-3.3
--- buildroot/target/linux/ar71xx/config-3.3	2014-04-23 21:33:04.000000000 +0200
+++ buildroot.c/target/linux/ar71xx/config-3.3	2014-05-31 22:59:24.420814564 +0200
@@ -213,6 +213,7 @@
 CONFIG_SOC_AR913X=y
 CONFIG_SOC_AR933X=y
 CONFIG_SOC_AR934X=y
+CONFIG_SOC_QCA953X=y
 CONFIG_SOC_QCA955X=y
 CONFIG_SPI=y
 CONFIG_SPI_AP83=y
diff -Naur --suppress-blank-empty --strip-trailing-cr buildroot/target/linux/ar71xx/files/arch/mips/ath79/dev-eth.c buildroot.c/target/linux/ar71xx/files/arch/mips/ath79/dev-eth.c
--- buildroot/target/linux/ar71xx/files/arch/mips/ath79/dev-eth.c	2014-04-23 21:33:04.000000000 +0200
+++ buildroot.c/target/linux/ar71xx/files/arch/mips/ath79/dev-eth.c	2014-05-31 23:00:50.992460582 +0200
@@ -195,6 +195,7 @@
 	case ATH79_SOC_AR7241:
 	case ATH79_SOC_AR9330:
 	case ATH79_SOC_AR9331:
+	case ATH79_SOC_QCA9533:
 		mdio_dev = &ath79_mdio1_device;
 		mdio_data = &ath79_mdio1_data;
 		break;
@@ -250,6 +251,12 @@
 		}
 		mdio_data->is_ar934x = 1;
 		break;
+
+	case ATH79_SOC_QCA9533:
+		mdio_data->builtin_switch = 1;
+		break;
+
+
 	case ATH79_SOC_QCA9558:
 		if (id == 1)
 			mdio_data->builtin_switch = 1;
@@ -540,6 +547,7 @@
 	case ATH79_SOC_AR9341:
 	case ATH79_SOC_AR9342:
 	case ATH79_SOC_AR9344:
+	case ATH79_SOC_QCA9533:
 	case ATH79_SOC_QCA9558:
 		pll_10 = AR934X_PLL_VAL_10;
 		pll_100 = AR934X_PLL_VAL_100;
@@ -596,6 +604,7 @@
 		case ATH79_SOC_AR7241:
 		case ATH79_SOC_AR9330:
 		case ATH79_SOC_AR9331:
+		case ATH79_SOC_QCA9533:
 			pdata->phy_if_mode = PHY_INTERFACE_MODE_MII;
 			break;

@@ -645,6 +654,7 @@
 		case ATH79_SOC_AR7241:
 		case ATH79_SOC_AR9330:
 		case ATH79_SOC_AR9331:
+		case ATH79_SOC_QCA9533:
 			pdata->phy_if_mode = PHY_INTERFACE_MODE_GMII;
 			break;

@@ -914,6 +924,37 @@
 			pdata->fifo_cfg3 = 0x01f00140;
 		break;

+	case ATH79_SOC_QCA9533:
+		if (id == 0) {
+			pdata->reset_bit = AR933X_RESET_GE0_MAC |
+					   AR933X_RESET_GE0_MDIO;
+			pdata->set_speed = ath79_set_speed_dummy;
+
+			pdata->phy_mask = BIT(4);
+		} else {
+			pdata->reset_bit = AR933X_RESET_GE1_MAC |
+					   AR933X_RESET_GE1_MDIO;
+			pdata->set_speed = ath79_set_speed_dummy;
+
+			pdata->speed = SPEED_1000;
+			pdata->duplex = DUPLEX_FULL;
+			pdata->switch_data = &ath79_switch_data;
+
+			ath79_switch_data.phy_poll_mask |= BIT(4);
+		}
+
+		pdata->ddr_flush = ath79_ddr_no_flush;
+		pdata->has_gbit = 1;
+		pdata->is_ar724x = 1;
+
+		if (!pdata->fifo_cfg1)
+			pdata->fifo_cfg1 = 0x0010ffff;
+		if (!pdata->fifo_cfg2)
+			pdata->fifo_cfg2 = 0x015500aa;
+		if (!pdata->fifo_cfg3)
+			pdata->fifo_cfg3 = 0x01f00140;
+		break;
+
 	default:
 		BUG();
 	}
@@ -953,6 +994,7 @@
 		case ATH79_SOC_AR7241:
 		case ATH79_SOC_AR9330:
 		case ATH79_SOC_AR9331:
+		case ATH79_SOC_QCA9533:
 			pdata->mii_bus_dev = &ath79_mdio1_device.dev;
 			break;

