diff -Naur --suppress-blank-empty --strip-trailing-cr openwrt.ori/target/linux/lantiq/image/tp-link.mk openwrt/target/linux/lantiq/image/tp-link.mk
--- openwrt.ori/target/linux/lantiq/image/tp-link.mk	2023-05-25 10:32:40.004439343 +0200
+++ openwrt/target/linux/lantiq/image/tp-link.mk	2023-07-23 19:33:24.314214030 +0200
@@ -18,7 +18,10 @@
 endef

 define Device/tplink_tdw8970
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/lantiqTpLink)
   DEVICE_MODEL := TD-W8970
   DEVICE_VARIANT := v1
diff -Naur --suppress-blank-empty --strip-trailing-cr openwrt.ori/target/linux/lantiq/image/vr9.mk openwrt/target/linux/lantiq/image/vr9.mk
--- openwrt.ori/target/linux/lantiq/image/vr9.mk	2023-05-25 10:32:40.004439343 +0200
+++ openwrt/target/linux/lantiq/image/vr9.mk	2023-07-23 18:47:11.001411992 +0200
@@ -120,7 +120,10 @@
 endef

 define Device/avm_fritz3370-rev2-hynix
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/avm_fritz3370)
   DEVICE_MODEL := FRITZ!Box 3370
   DEVICE_VARIANT := Rev. 2 (Hynix NAND)
@@ -128,7 +131,10 @@
 TARGET_DEVICES += avm_fritz3370-rev2-hynix

 define Device/avm_fritz3370-rev2-micron
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/avm_fritz3370)
   DEVICE_MODEL := FRITZ!Box 3370
   DEVICE_VARIANT := Rev. 2 (Micron NAND)
@@ -148,7 +154,10 @@
 TARGET_DEVICES += avm_fritz3390

 define Device/avm_fritz7360sl
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/AVM)
   DEVICE_MODEL := FRITZ!Box 7360 SL
   IMAGE_SIZE := 15744k
@@ -158,7 +167,10 @@
 TARGET_DEVICES += avm_fritz7360sl

 define Device/avm_fritz7360-v2
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/AVM)
   DEVICE_MODEL := FRITZ!Box 7360
   DEVICE_VARIANT := v2
@@ -168,7 +180,10 @@
 TARGET_DEVICES += avm_fritz7360-v2

 define Device/avm_fritz7362sl
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/AVM)
   $(Device/NAND)
   DEVICE_MODEL := FRITZ!Box 7362 SL
@@ -179,7 +194,10 @@
 TARGET_DEVICES += avm_fritz7362sl

 define Device/avm_fritz7412
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/AVM)
   $(Device/NAND)
   DEVICE_MODEL := FRITZ!Box 7412
@@ -191,7 +209,10 @@
 TARGET_DEVICES += avm_fritz7412

 define Device/avm_fritz7430
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/AVM)
   $(Device/NAND)
   DEVICE_MODEL := FRITZ!Box 7430
diff -Naur --suppress-blank-empty --strip-trailing-cr openwrt.ori/target/linux/lantiq/xrx200/base-files/etc/board.d/05_compat-version openwrt/target/linux/lantiq/xrx200/base-files/etc/board.d/05_compat-version
--- openwrt.ori/target/linux/lantiq/xrx200/base-files/etc/board.d/05_compat-version	2023-05-25 10:31:23.668348832 +0200
+++ openwrt/target/linux/lantiq/xrx200/base-files/etc/board.d/05_compat-version	2023-07-23 19:35:06.214669829 +0200
@@ -8,6 +8,21 @@
 board_config_update

 case "$(board_name)" in
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+# only does with dsa migration in:
+#  lantiq/image/vr9.mk.mk
+#  lantiq/image/tp-link.mk
+	avm,fritz3370-rev2-hynix|\
+	avm,fritz3370-rev2-micron|\
+	avm,fritz7360sl|\
+	avm,fritz7360-v2|\
+	avm,fritz7362sl|\
+	avm,fritz7412|\
+	avm,fritz7430|\
+	tplink,tdw8970)
+		;;
 	*)
 		ucidef_set_compat_version "1.1"
 		;;
diff -Naur --suppress-blank-empty --strip-trailing-cr openwrt.ori/target/linux/ramips/image/mt7621.mk openwrt/target/linux/ramips/image/mt7621.mk
--- openwrt.ori/target/linux/ramips/image/mt7621.mk	2023-05-25 10:32:40.044439390 +0200
+++ openwrt/target/linux/ramips/image/mt7621.mk	2023-07-23 21:38:19.967026937 +0200
@@ -520,7 +520,10 @@
 TARGET_DEVICES += dual-q_h721

 define Device/d-team_newifi-d2
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 32448k
   DEVICE_VENDOR := Newifi
@@ -718,7 +721,10 @@
 TARGET_DEVICES += gehua_ghl-r-001

 define Device/glinet_gl-mt1300
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   IMAGE_SIZE := 32448k
   DEVICE_VENDOR := GL.iNet
   DEVICE_MODEL := GL-MT1300
@@ -1239,7 +1245,10 @@
 TARGET_DEVICES += mqmaker_witi

 define Device/mtc_wr1201
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 16000k
   DEVICE_VENDOR := MTC
@@ -1285,7 +1294,6 @@
 TARGET_DEVICES += netgear_ex6150

 define Device/netgear_sercomm_nand
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   BLOCKSIZE := 128k
   PAGESIZE := 2048
@@ -1303,6 +1311,10 @@
 endef

 define Device/netgear_r6220
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6220
   SERCOMM_HWNAME := R6220
@@ -1317,6 +1329,7 @@


 define Device/netgear_r6260
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6260
   SERCOMM_HWNAME := R6260
@@ -1329,6 +1342,7 @@
 TARGET_DEVICES += netgear_r6260

 define Device/netgear_r6350
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6350
   SERCOMM_HWNAME := R6350
@@ -1341,6 +1355,7 @@
 TARGET_DEVICES += netgear_r6350

 define Device/netgear_r6700-v2
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6700
   DEVICE_VARIANT := v2
@@ -1360,6 +1375,7 @@
 TARGET_DEVICES += netgear_r6700-v2

 define Device/netgear_r6800
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6800
   SERCOMM_HWNAME := R6950
@@ -1372,6 +1388,7 @@
 TARGET_DEVICES += netgear_r6800

 define Device/netgear_r6850
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6850
   SERCOMM_HWNAME := R6850
@@ -1384,6 +1401,7 @@
 TARGET_DEVICES += netgear_r6850

 define Device/netgear_r6900-v2
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R6900
   DEVICE_VARIANT := v2
@@ -1397,6 +1415,7 @@
 TARGET_DEVICES += netgear_r6900-v2

 define Device/netgear_r7200
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R7200
   SERCOMM_HWNAME := R6950
@@ -1409,6 +1428,7 @@
 TARGET_DEVICES += netgear_r7200

 define Device/netgear_r7450
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := R7450
   SERCOMM_HWNAME := R6950
@@ -1421,6 +1441,10 @@
 TARGET_DEVICES += netgear_r7450

 define Device/netgear_wac104
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := WAC104
   SERCOMM_HWNAME := WAC104
@@ -1433,6 +1457,7 @@
 TARGET_DEVICES += netgear_wac104

 define Device/netgear_wac124
+  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nand)
   DEVICE_MODEL := WAC124
   SERCOMM_HWNAME := WAC124
@@ -1805,7 +1830,10 @@
 TARGET_DEVICES += tplink_tl-wpa8631p-v3

 define Device/ubnt_edgerouter_common
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   DEVICE_VENDOR := Ubiquiti
   IMAGE_SIZE := 256768k
@@ -2052,7 +2080,10 @@
 TARGET_DEVICES += xiaomi_mi-router-4

 define Device/xiaomi_mi-router-4a-gigabit
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 14848k
   DEVICE_VENDOR := Xiaomi
@@ -2132,7 +2163,10 @@
 TARGET_DEVICES += xzwifi_creativebox-v1

 define Device/youhua_wr1200js
-  $(Device/dsa-migration)
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+#  $(Device/dsa-migration)
   IMAGE_SIZE := 16064k
   DEVICE_VENDOR := YouHua
   DEVICE_MODEL := WR1200JS
diff -Naur --suppress-blank-empty --strip-trailing-cr openwrt.ori/target/linux/ramips/mt7621/base-files/etc/board.d/05_compat-version openwrt/target/linux/ramips/mt7621/base-files/etc/board.d/05_compat-version
--- openwrt.ori/target/linux/ramips/mt7621/base-files/etc/board.d/05_compat-version	2023-05-25 10:31:23.732348910 +0200
+++ openwrt/target/linux/ramips/mt7621/base-files/etc/board.d/05_compat-version	2023-07-23 21:35:25.286086014 +0200
@@ -8,6 +8,19 @@
 board_config_update

 case "$(board_name)" in
+# ddmesh: disabled because we can upgrade to 22
+#         The new version breaks sysupgrade for existing devices.
+#         meaning, that this version for dsa will not be used.
+# only does with dsa migration in mt7621.mk
+	glinet,gl-mt1300|\
+	mtc,wr1201|\
+	d-team,newifi-d2|\
+	xiaomi,mi-router-4a-gigabit|\
+	youhua,wr1200js|\
+	ubnt,edgerouter-x|\
+	netgear,wac104|\
+	netgear,r6220)
+		;;
 	*)
 		ucidef_set_compat_version "1.1"
 		;;
