--- buildroot/package/utils/busybox/Config-defaults.in.original	2019-09-15 19:17:29.811216595 +0200
+++ buildroot/package/utils/busybox/Config-defaults.in	2019-09-15 19:29:50.381681392 +0200
@@ -1309,7 +1309,7 @@
 	default 0
 config BUSYBOX_DEFAULT_CHPASSWD
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_FEATURE_DEFAULT_PASSWD_ALGO
 	string
 	default "md5"
@@ -1351,7 +1351,7 @@
 	default y
 config BUSYBOX_DEFAULT_FEATURE_PASSWD_WEAK_CHECK
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_SU
 	bool
 	default n
@@ -1648,7 +1648,7 @@
 	default n
 config BUSYBOX_DEFAULT_MKSWAP
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_MKSWAP_UUID
 	bool
 	default n
@@ -1675,7 +1675,7 @@
 	default n
 config BUSYBOX_DEFAULT_FEATURE_MOUNT_CIFS
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_MOUNT_FLAGS
 	bool
 	default y
@@ -1750,16 +1750,16 @@
 	default n
 config BUSYBOX_DEFAULT_SWAPON
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_SWAPON_DISCARD
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_SWAPON_PRI
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_SWAPOFF
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_SWAPONOFF_LABEL
 	bool
 	default n
@@ -2360,7 +2360,7 @@
 	default "/etc/iproute2"
 config BUSYBOX_DEFAULT_FEATURE_IP_TUNNEL
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_FEATURE_IP_RULE
 	bool
 	default y
@@ -2735,7 +2735,7 @@
 	default n
 config BUSYBOX_DEFAULT_PSTREE
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_PWDX
 	bool
 	default n
@@ -2882,7 +2882,7 @@
 	default y
 config BUSYBOX_DEFAULT_ASH_OPTIMIZE_FOR_SIZE
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_ASH_INTERNAL_GLOB
 	bool
 	default y
