--- buildroot/include/target.mk.orig	2019-09-07 13:07:56.614770206 +0200
+++ buildroot/include/target.mk	2019-09-07 13:53:17.469045344 +0200
@@ -17,7 +17,7 @@
 # For nas targets
 DEFAULT_PACKAGES.nas:=block-mount fdisk lsblk mdadm
 # For router targets
-DEFAULT_PACKAGES.router:=dnsmasq iptables ip6tables ppp ppp-mod-pppoe firewall odhcpd-ipv6only odhcp6c kmod-ipt-offload
+DEFAULT_PACKAGES.router:=dnsmasq iptables firewall
 DEFAULT_PACKAGES.bootloader:=

 ifneq ($(DUMP),)
