--- buildroot/config/Config-kernel.in.orig	2019-09-07 13:07:50.762744079 +0200
+++ buildroot/config/Config-kernel.in	2019-09-07 13:33:35.980283003 +0200
@@ -29,11 +29,11 @@
 config KERNEL_CRASHLOG
 	bool "Crash logging"
 	depends on !(arm || powerpc || sparc || TARGET_uml || i386 || x86_64)
-	default y
+	default n
 
 config KERNEL_SWAP
 	bool "Support for paging of anonymous memory (swap)"
-	default y if !SMALL_FLASH
+	default n
 
 config KERNEL_DEBUG_FS
 	bool "Compile the kernel with debug filesystem enabled"
@@ -90,7 +90,7 @@
 
 config KERNEL_KALLSYMS
 	bool "Compile the kernel with symbol table information"
-	default y if !SMALL_FLASH
+	default n
 	help
 	  This will give you more information in stack traces from kernel oopses.
 
@@ -135,7 +135,7 @@
 
 config KERNEL_DEBUG_INFO
 	bool "Compile the kernel with debug information"
-	default y if !SMALL_FLASH
+	default n
 	select KERNEL_DEBUG_KERNEL
 	help
 	  This will compile your kernel and modules with debug information.
@@ -216,7 +216,7 @@
 
 config KERNEL_MAGIC_SYSRQ
 	bool "Compile the kernel with SysRq support"
-	default y
+	default n
 
 config KERNEL_DEBUG_PINCTRL
 	bool "Compile the kernel with pinctrl debugging"
@@ -228,11 +228,12 @@
 
 config KERNEL_COREDUMP
 	bool
+	default n
 
 config KERNEL_ELF_CORE
 	bool "Enable process core dump support"
 	select KERNEL_COREDUMP
-	default y if !SMALL_FLASH
+	default n
 
 config KERNEL_PROVE_LOCKING
 	bool "Enable kernel lock checking"
@@ -837,7 +838,7 @@
 #
 choice
 	prompt "Compiler optimization level"
-	default KERNEL_CC_OPTIMIZE_FOR_SIZE if SMALL_FLASH
+	default KERNEL_CC_OPTIMIZE_FOR_SIZE
 
 config KERNEL_CC_OPTIMIZE_FOR_PERFORMANCE
 	bool "Optimize for performance"
@@ -848,6 +849,7 @@

 config KERNEL_CC_OPTIMIZE_FOR_SIZE
 	bool "Optimize for size"
+	default y
 	help
	  Enabling this option will pass "-Os" instead of "-O2" to
	  your compiler resulting in a smaller kernel.
