--- buildroot/package/base-files/image-config.in.ori	2020-10-07 00:14:39.409837748 +0000
+++ buildroot/package/base-files/image-config.in	2020-10-07 00:16:48.938575609 +0000
@@ -183,7 +183,7 @@
 	config VERSION_REPO
 		string
 		prompt "Release repository"
-		default "http://downloads.openwrt.org/releases/19.07.4"
+		default "https://github.com/freifunk-dresden/firmware-freifunk-dresden.git"
 		help
 			This is the repository address embedded in the image, it defaults
 			to the trunk snapshot repo; the url may contain the following placeholders:
@@ -210,7 +210,7 @@

 	config VERSION_MANUFACTURER
 		string
-		prompt "Manufacturer name"
+		prompt "Freifunk Dresden"
 		help
 			This is the manufacturer name embedded in /etc/device_info
 			Useful for OEMs building OpenWrt based firmware
@@ -251,7 +251,7 @@
 	config VERSION_FILENAMES
 		bool
 		prompt "Version number in filenames"
-		default y
+		default n
 		help
 			Enable this to include the version number in firmware image, SDK-
 			and Image Builder archive file names
@@ -269,7 +269,7 @@

 menuconfig PER_FEED_REPO
 	bool "Separate feed repositories" if IMAGEOPT
-	default y
+	default n
 	help
 		If set, a separate repository is generated within bin/*/packages/
 		for the core packages and each enabled feed.
