--- buildroot/Makefile	2020-01-25 16:20:20.409756680 +0000
+++ buildroot/Makefile	2020-01-25 16:23:24.694621713 +0000
@@ -54,6 +54,9 @@

 prepare: $(target/stamp-compile)

+lightclean: FORCE
+	rm -rf $(BUILD_DIR) $(STAGING_DIR) $(BUILD_LOG_DIR) $(TOPDIR)/staging_dir/packages
+
 clean: FORCE
 	rm -rf $(BUILD_DIR) $(STAGING_DIR) $(BIN_DIR) $(OUTPUT_DIR)/packages/$(ARCH_PACKAGES) $(BUILD_LOG_DIR) $(TOPDIR)/staging_dir/packages

