--- a/package/firmware/ath10k-ct-firmware/Makefile
+++ b/package/firmware/ath10k-ct-firmware/Makefile
@@ -91,7 +91,7 @@ endef
 QCA988X_FIRMWARE_FILE_CT:=firmware-2-ct-full-community-22.bin.lede.019
 define Download/ath10k-firmware-qca988x-ct
   $(call Download/ct-firmware,QCA988X,)
-  HASH:=8b4c99253aa309d35f2e060c190091b8db1b84dbda06a6a15c83ac0f9a938126
+  HASH:=072fafbd4d741ba77f5d92b60a644c7360af0c21a3a9ec06f1c299484bf41688
 endef
 $(eval $(call Download,ath10k-firmware-qca988x-ct))
 
@@ -98,7 +98,7 @@ endef
 QCA988X_FIRMWARE_FILE_CT_HTT:=firmware-2-ct-full-htt-mgt-community-22.bin.lede.019
 define Download/ath10k-firmware-qca988x-ct-htt
   $(call Download/ct-firmware-htt,QCA988X,)
-  HASH:=a7168916d6aa5e4d7858f8b620c0c980c76d03f390929db6f4077685ce2051e7
+  HASH:=031ee5eb9704a5df51d7fa25f326e334205f7bae8bfcf34327ee82d7671ee7a4
 endef
 $(eval $(call Download,ath10k-firmware-qca988x-ct-htt))
 
@@ -121,7 +121,7 @@ endef
 QCA99X0_FIRMWARE_FILE_CT:=firmware-5-ct-full-community-12.bin-lede.019
 define Download/ath10k-firmware-qca99x0-ct
   $(call Download/ct-firmware,QCA99X0,ath10k-10-4b)
-  HASH:=7dc934f934bc4973c9273a4f22cfead8e26ec6f579647af31b718a860eca0a4b
+  HASH:=4abdc7c63abb19a54b5021fc5b077cf33f97743e73ac2c0d3c6befa84e86f81a
 endef
 $(eval $(call Download,ath10k-firmware-qca99x0-ct))
 
@@ -128,7 +128,7 @@ endef
 QCA99X0_FIRMWARE_FILE_CT_HTT:=firmware-5-ct-full-htt-mgt-community-12.bin-lede.019
 define Download/ath10k-firmware-qca99x0-ct-htt
   $(call Download/ct-firmware-htt,QCA99X0,ath10k-10-4b)
-  HASH:=71a27b245a382fe009938d2826d5c97a90dceb10ddf638325268df91837ea302
+  HASH:=ac3aa9a932f4afcdbf065b1214151593b2ff59e3bd5779bdea530a6380307193
 endef
 $(eval $(call Download,ath10k-firmware-qca99x0-ct-htt))
 
@@ -136,7 +136,7 @@ endef
 QCA9984_FIRMWARE_FILE_CT:=firmware-5-ct-full-community-12.bin-lede.019
 define Download/ath10k-firmware-qca9984-ct
   $(call Download/ct-firmware,QCA9984,ath10k-9984-10-4b)
-  HASH:=32d13f432691fe759ded7d027052e925233adb436cd8f729f85ec3d19ccd1dfd
+  HASH:=dfa2a6fe4fbcc481abd014c15876afee526ffbc29a7a73af5a86ddaa161cabcf
 endef
 $(eval $(call Download,ath10k-firmware-qca9984-ct))
 
@@ -143,7 +143,7 @@ endef
 QCA9984_FIRMWARE_FILE_CT_HTT:=firmware-5-ct-full-htt-mgt-community-12.bin-lede.019
 define Download/ath10k-firmware-qca9984-ct-htt
   $(call Download/ct-firmware-htt,QCA9984,ath10k-9984-10-4b)
-  HASH:=e8ab69777bd00b5fc6b1b7acccb55b903553a99932a5b0351602b5f690106588
+  HASH:=0a44560ad2ee0cf22547ec3e2abfd5820446b6fb9d03a8effd2717885f6700e9
 endef
 $(eval $(call Download,ath10k-firmware-qca9984-ct-htt))
 
@@ -151,7 +151,7 @@ endef
 QCA4019_FIRMWARE_FILE_CT:=firmware-5-ct-full-community-12.bin-lede.019
 define Download/ath10k-firmware-qca4019-ct
   $(call Download/ct-firmware,QCA4019,ath10k-4019-10-4b)
-  HASH:=4b89763087c7ed9b56046c4e621b7f045e452436d8d9b430a5d171179e313592
+  HASH:=7d9df54167b3feb770f967091ec6f33e7bec79be47df09bea36a2d55d6a661d9
 endef
 $(eval $(call Download,ath10k-firmware-qca4019-ct))
 
@@ -158,7 +158,7 @@ endef
 QCA4019_FIRMWARE_FILE_CT_HTT:=firmware-5-ct-full-htt-mgt-community-12.bin-lede.019
 define Download/ath10k-firmware-qca4019-ct-htt
   $(call Download/ct-firmware-htt,QCA4019,ath10k-4019-10-4b)
-  HASH:=fba591e5777c53b82542ba16cae69d9bb4684837f2fa4cee1b9b26f648096748
+  HASH:=98fb0b2168c14e1ed8bce1c0f30bc42fa137cc00be38caf32a55b1cfa9334b05
 endef
 $(eval $(call Download,ath10k-firmware-qca4019-ct-htt))
 
@@ -166,7 +166,7 @@ endef
 QCA9888_FIRMWARE_FILE_CT:=firmware-5-ct-full-community-12.bin-lede.019
 define Download/ath10k-firmware-qca9888-ct
   $(call Download/ct-firmware,QCA9888,ath10k-9888-10-4b)
-  HASH:=048f4300725e6ebbf94a6bf4f3f4e4592c446fcdbe1d801aaac024b15e89e0c9
+  HASH:=94d8902bf09cab73b768938a1947de4703ed0267a333219ceec4fe852f53527f
 endef
 $(eval $(call Download,ath10k-firmware-qca9888-ct))
 
@@ -173,7 +173,7 @@ endef
 QCA9888_FIRMWARE_FILE_CT_HTT:=firmware-5-ct-full-htt-mgt-community-12.bin-lede.019
 define Download/ath10k-firmware-qca9888-ct-htt
   $(call Download/ct-firmware-htt,QCA9888,ath10k-9888-10-4b)
-  HASH:=d2a7e9fea6bd854721b3fc03a3a00d379d303b2bce339377ee87a1c14a60312d
+  HASH:=99e58cb63d150b456bc70e77665fe64d8a7193a96ee11ec0ef55b6f8c32ea39f
 endef
 $(eval $(call Download,ath10k-firmware-qca9888-ct-htt))
