--- buildroot/package/boot/uboot-envtools/files/ar71xx
+++ buildroot/package/boot/uboot-envtools/files/ar71xx
@@ -37,6 +37,7 @@ gl-ar300m|\
 gl-ar750|\
 hornet-ub|\
 hornet-ub-x2|\
+jt-or750i|\
 jwap230|\
 koala|\
 mr1750|\
diff --git a/target/linux/ar71xx/base-files/etc/board.d/01_leds b/target/linux/ar71xx/base-files/etc/board.d/01_leds
index 23f3d23bf53883fb08944bc65b98cac06a16c0a5..c5f863765a01e4595c535615dd8fc2f124f02cb0 100755
--- buildroot/target/linux/ar71xx/base-files/etc/board.d/01_leds
+++ buildroot/target/linux/ar71xx/base-files/etc/board.d/01_leds
@@ -111,14 +111,20 @@ n5q)
 		;;
 	esac
 	;;
-archer-c25-v1)
+archer-c25-v1|\
+jt-or750i)	
 	ucidef_set_led_netdev "wan" "WAN" "$board:green:wan" "eth0"
-	ucidef_set_led_wlan "wlan" "WLAN" "$board:green:wlan2g" "phy1tpt"
-	ucidef_set_led_wlan "wlan5g" "WLAN5G" "$board:green:wlan5g" "phy0tpt"
 	ucidef_set_led_switch "lan1" "LAN1" "$board:green:lan1" "switch0" "0x10"
 	ucidef_set_led_switch "lan2" "LAN2" "$board:green:lan2" "switch0" "0x08"
 	ucidef_set_led_switch "lan3" "LAN3" "$board:green:lan3" "switch0" "0x04"
-	ucidef_set_led_switch "lan4" "LAN4" "$board:green:lan4" "switch0" "0x02"
+
+	case "$board" in
+	archer-c25-v1)
+		ucidef_set_led_switch "lan4" "LAN4" "$board:green:lan4" "switch0" "0x02"
+		ucidef_set_led_wlan "wlan" "WLAN" "$board:green:wlan2g" "phy1tpt"
+		ucidef_set_led_wlan "wlan5g" "WLAN5G" "$board:green:wlan5g" "phy0tpt"
+		;;
+	esac
 	;;
 archer-c5|\
 archer-c7)
diff --git a/target/linux/ar71xx/base-files/etc/board.d/02_network b/target/linux/ar71xx/base-files/etc/board.d/02_network
index e9522252a2b8ba7de1c8ca7d070ff7f2897aade9..c4838e8ec593f1fb166602e012e43a24443e47a2 100755
--- buildroot/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ buildroot/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -475,6 +475,11 @@
 		ucidef_add_switch "switch0" \
 			"0:lan:4" "1:lan:3" "2:lan:2" "3:lan:1" "5@eth1"
 		;;
+	jt-or750i)
+		ucidef_set_interfaces_lan_wan "eth1" "eth0"
+		ucidef_add_switch "switch0" \
+			"0@eth1" "2:lan:3" "3:lan:2" "4:lan:1"
+		;;
 	routerstation-pro)
 		ucidef_set_interfaces_lan_wan "eth1.1" "eth0"
 		ucidef_add_switch "switch0" \
diff --git a/target/linux/ar71xx/base-files/etc/diag.sh b/target/linux/ar71xx/base-files/etc/diag.sh
index 4eb2abb26ca2676576babb14eb1baa38a2f0de2b..7a52bcac305d635eb22db409fce1a73242ec64d5 100644
--- buildroot/target/linux/ar71xx/base-files/etc/diag.sh
+++ buildroot/target/linux/ar71xx/base-files/etc/diag.sh
@@ -263,6 +263,9 @@ get_status_led() {
 	ja76pf2)
 		status_led="jjplus:green:led1"
 		;;
+	jt-or750i)
+		status_led="$board:inv_red:status"
+		;;
 	jwap230)
 		status_led="$board:green:led1"
 		;;
diff --git a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index 2ba8de4783..077f28768f 100644
--- buildroot/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ buildroot/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -76,6 +76,7 @@ case "$FIRMWARE" in
 	dlan-pro-1200-ac|\
 	e1700ac-v2|\
 	e600gac-v2|\
+	jt-or750i|\
 	minibox-v3.2|\
 	oolite-v5.2|\
 	oolite-v5.2-dev|\
diff --git a/target/linux/ar71xx/base-files/lib/ar71xx.sh b/target/linux/ar71xx/base-files/lib/ar71xx.sh
index ccbd4e77c324a36e7fba6e6dccad59d8f94a3921..5b5130e97dd61b20e1566d1e196f164af33b3467 100755
--- buildroot/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ buildroot/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -792,6 +792,9 @@ ar71xx_board_detect() {
 	*"JA76PF2")
 		name="ja76pf2"
 		;;
+	*"JT-OR750i")
+		name="jt-or750i"
+		;;
 	*"JWAP003")
 		name="jwap003"
 		;;
diff --git a/target/linux/ar71xx/base-files/lib/upgrade/platform.sh b/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
index ca1270b7fdc84c53f8417c226f18ca4fff1f27d8..65ed1d8b7ada9cfba4b7d1e8b72d17594aa5e1b6 100755
--- buildroot/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
+++ buildroot/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
@@ -272,6 +272,7 @@ platform_check_image() {
 	gl-usb150|\
 	hiwifi-hc6361|\
 	hornet-ub-x2|\
+	jt-or750i|\
 	jwap230|\
 	lbe-m5|\
 	lima|\
diff --git a/target/linux/ar71xx/config-4.14 b/target/linux/ar71xx/config-4.14
index 9a524fae43..a4ca5ff699 100644
--- buildroot/target/linux/ar71xx/config-4.14
+++ buildroot/target/linux/ar71xx/config-4.14
@@ -137,6 +137,7 @@ CONFIG_ATH79=y
 # CONFIG_ATH79_MACH_HIWIFI_HC6361 is not set
 # CONFIG_ATH79_MACH_HORNET_UB is not set
 # CONFIG_ATH79_MACH_JA76PF is not set
+# CONFIG_ATH79_MACH_JT_OR750I is not set
 # CONFIG_ATH79_MACH_JWAP003 is not set
 # CONFIG_ATH79_MACH_JWAP230 is not set
 # CONFIG_ATH79_MACH_KOALA is not set
diff --git a/target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt b/target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt
index 1d0e3d910d42ab87af8481d6d13a79949d68e41b..5f2696ad424aef9b1cd300c2bcd3111e7fad3935 100644
--- buildroot/target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt
+++ buildroot/target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt
@@ -965,6 +965,17 @@ config ATH79_MACH_JA76PF
 	select ATH79_DEV_M25P80
 	select ATH79_DEV_USB
 
+config ATH79_MACH_JT_OR750I
+	bool "Joy-IT OR750i support"
+	select SOC_QCA953X
+	select ATH79_DEV_AP9X_PCI if PCI
+	select ARH79_DEV_ETH
+	select ARH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+
 config ATH79_MACH_JWAP003
 	bool "jjPlus JWAP003 board support"
 	select SOC_AR71XX
diff --git a/target/linux/ar71xx/files/arch/mips/ath79/Makefile b/target/linux/ar71xx/files/arch/mips/ath79/Makefile
index 8800a7de1020062cbf4f25cf507ed9b1fbe78da0..8a992507ff4f6ac77bd64743c927f29d4d2cc1d8 100644
--- buildroot/target/linux/ar71xx/files/arch/mips/ath79/Makefile
+++ buildroot/target/linux/ar71xx/files/arch/mips/ath79/Makefile
@@ -145,6 +145,7 @@ obj-$(CONFIG_ATH79_MACH_HIVEAP_121)		+= mach-hiveap-121.o
 obj-$(CONFIG_ATH79_MACH_HIWIFI_HC6361)		+= mach-hiwifi-hc6361.o
 obj-$(CONFIG_ATH79_MACH_HORNET_UB)		+= mach-hornet-ub.o
 obj-$(CONFIG_ATH79_MACH_JA76PF)			+= mach-ja76pf.o
+obj-$(CONFIG_ATH79_MACH_JT_OR750I)		+= mach-jt-or750i.o
 obj-$(CONFIG_ATH79_MACH_JWAP003)		+= mach-jwap003.o
 obj-$(CONFIG_ATH79_MACH_JWAP230)		+= mach-jwap230.o
 obj-$(CONFIG_ATH79_MACH_KOALA)			+= mach-koala.o
diff --git a/target/linux/ar71xx/files/arch/mips/ath79/mach-jt-or750i.c b/target/linux/ar71xx/files/arch/mips/ath79/mach-jt-or750i.c
new file mode 100644
index 0000000000000000000000000000000000000000..7ce8abb4d21a3cbb61ecec784ae7f177505e1bca
--- /dev/null
+++ buildroot/target/linux/ar71xx/files/arch/mips/ath79/mach-jt-or750i.c
@@ -0,0 +1,122 @@
+/*
+ * Joy-IT JT-OR750i board support
+ *
+ * Copyright (C) 2019 Vincent Wiemann <vcw@derowe.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation.
+ */
+
+#include <linux/gpio.h>
+#include <linux/platform_device.h>
+
+#include <asm/mach-ath79/ath79.h>
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-usb.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+#include "pci.h"
+
+/* ath10k GPIO is not used as a WLAN-LED, but as a status LED with default
+ * state "on" together with the inverted red status LED for signalling e.g.
+ * failsafe mode
+ */
+#define OR750I_GPIO_LED_STATUS_GREEN	506
+#define OR750I_GPIO_LED_STATUS_RED	13
+#define OR750I_GPIO_LED_LAN1		16
+#define OR750I_GPIO_LED_LAN2		15
+#define OR750I_GPIO_LED_LAN3		14
+#define OR750I_GPIO_LED_WAN		4
+
+#define OR750I_GPIO_BTN_RESET		17
+
+#define OR750I_KEYS_POLL_INTERVAL	20 /* msec */
+#define OR750I_KEYS_DEBOUNCE_INTERVAL	(3 * OR750I_KEYS_POLL_INTERVAL)
+
+#define OR750I_WMAC2G_CALDATA_OFFSET	0x1000
+#define OR750I_WMAC5G_CALDATA_OFFSET	0x5000
+
+static struct gpio_led or750i_gpio_leds[] __initdata = {
+	{
+		.name		= "jt-or750i:green:lan1",
+		.gpio		= OR750I_GPIO_LED_LAN1,
+		.active_low	= 1,
+	}, {
+		.name		= "jt-or750i:green:lan2",
+		.gpio		= OR750I_GPIO_LED_LAN2,
+		.active_low	= 1,
+	}, {
+		.name		= "jt-or750i:green:lan3",
+		.gpio		= OR750I_GPIO_LED_LAN3,
+		.active_low	= 1,
+	}, {
+		.name		= "jt-or750i:green:wan",
+		.gpio		= OR750I_GPIO_LED_WAN,
+		.active_low	= 1,
+	}, {	/* Inverted active_low, because a red status LED looks broken */
+		.name		= "jt-or750i:inv_red:status",
+		.gpio		= OR750I_GPIO_LED_STATUS_RED,
+		.active_low	= 0,
+		.default_state	= 0,
+	}
+};
+
+static struct gpio_keys_button or750i_gpio_keys[] __initdata = {
+	{
+		.desc			= "reset",
+		.type			= EV_KEY,
+		.code			= KEY_RESTART,
+		.debounce_interval	= OR750I_KEYS_DEBOUNCE_INTERVAL,
+		.gpio			= OR750I_GPIO_BTN_RESET,
+		.active_low		= 1,
+	}
+};
+
+static void __init or750i_setup(void)
+{
+	u8 *art = (u8 *) KSEG1ADDR(0x1fff0000);
+
+	ath79_register_m25p80(NULL);
+
+	ath79_setup_ar933x_phy4_switch(false, false);
+
+	ath79_register_mdio(0, 0x0);
+
+	/* WAN */
+	ath79_init_mac(ath79_eth0_data.mac_addr, art, 0);
+	ath79_eth0_data.duplex = DUPLEX_FULL;
+	ath79_eth0_data.phy_if_mode = PHY_INTERFACE_MODE_MII;
+	ath79_eth0_data.phy_mask = BIT(4);
+	ath79_eth0_data.speed = SPEED_100;
+	ath79_register_eth(0);
+
+	/* LAN */
+	ath79_init_mac(ath79_eth1_data.mac_addr, art + 6, 0);
+	ath79_eth1_data.duplex = DUPLEX_FULL;
+	ath79_eth1_data.phy_if_mode = PHY_INTERFACE_MODE_GMII;
+	ath79_switch_data.phy4_mii_en = 1;
+	ath79_switch_data.phy_poll_mask |= BIT(4);
+	ath79_register_eth(1);
+
+	ath79_register_gpio_keys_polled(-1, OR750I_KEYS_POLL_INTERVAL,
+					ARRAY_SIZE(or750i_gpio_keys),
+					or750i_gpio_keys);
+
+	ath79_register_leds_gpio(-1, ARRAY_SIZE(or750i_gpio_leds),
+				 or750i_gpio_leds);
+
+	ath79_register_usb();
+
+	ath79_register_wmac(art + OR750I_WMAC2G_CALDATA_OFFSET, NULL);
+
+	ath79_register_pci();
+}
+
+MIPS_MACHINE(ATH79_MACH_JT_OR750I, "JT-OR750I", "Joy-IT JT-OR750i", or750i_setup);
diff --git a/target/linux/ar71xx/files/arch/mips/ath79/machtypes.h b/target/linux/ar71xx/files/arch/mips/ath79/machtypes.h
index 80f6e1d95b7a4e5559e7d5da041b32d962ad4e84..65005c19741359f492e4e23ea71826c1ba20071a 100644
--- buildroot/target/linux/ar71xx/files/arch/mips/ath79/machtypes.h
+++ buildroot/target/linux/ar71xx/files/arch/mips/ath79/machtypes.h
@@ -144,6 +144,7 @@ enum ath79_mach_type {
 	ATH79_MACH_HORNET_UB,			/* ALFA Networks Hornet-UB */
 	ATH79_MACH_JA76PF,			/* jjPlus JA76PF */
 	ATH79_MACH_JA76PF2,			/* jjPlus JA76PF2 */
+	ATH79_MACH_JT_OR750I,			/* Joy-IT JT-OR750i */
 	ATH79_MACH_JWAP003,			/* jjPlus JWAP003 */
 	ATH79_MACH_JWAP230,			/* jjPlus JWAP230 */
 	ATH79_MACH_KOALA,			/* OCEDO Koala */
diff --git a/target/linux/ar71xx/generic/config-default b/target/linux/ar71xx/generic/config-default
index 55ccd93fedbcf3062c94f18ec57dc132b272a83c..036fcdd4c96eebb2f7953011bf5e96bbba512d64 100644
--- buildroot/target/linux/ar71xx/generic/config-default
+++ buildroot/target/linux/ar71xx/generic/config-default
@@ -106,6 +106,7 @@ CONFIG_ATH79_MACH_GS_OOLITE_V5_2=y
 CONFIG_ATH79_MACH_HIWIFI_HC6361=y
 CONFIG_ATH79_MACH_HORNET_UB=y
 CONFIG_ATH79_MACH_JA76PF=y
+CONFIG_ATH79_MACH_JT_OR750I=y
 CONFIG_ATH79_MACH_JWAP003=y
 CONFIG_ATH79_MACH_JWAP230=y
 CONFIG_ATH79_MACH_KOALA=y
diff --git a/target/linux/ar71xx/image/generic.mk b/target/linux/ar71xx/image/generic.mk
index 4568b656219419e9ca1156c6716bd2124074cb32..4a73c47a8abc530dec2484c937a105a88a2dbf98 100644
--- buildroot/target/linux/ar71xx/image/generic.mk
+++ buildroot/target/linux/ar71xx/image/generic.mk
@@ -551,6 +551,18 @@ define Device/gl-usb150
 endef
 TARGET_DEVICES += gl-usb150
 
+define Device/jt-or750i
+  DEVICE_TITLE := Joy-IT JT-OR750i
+  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca9887 kmod-usb-core kmod-usb2
+  BOARDNAME := JT-OR750I
+  IMAGE_SIZE := 16000k
+  MTDPARTS := spi0.0:256k(u-boot)ro,64k(u-boot-env),16000k(firmware),64k(art)ro
+  SUPPORTED_DEVICES := jt-or750i
+  IMAGE/sysupgrade.bin := append-kernel | pad-to $$$$(BLOCKSIZE) |\
+	append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
+endef
+TARGET_DEVICES += jt-or750i
+
 define Device/lan-turtle
   $(Device/tplink-16mlzma)
   DEVICE_TITLE := Hak5 LAN Turtle
