diff -Naur mac80211.or/Makefile mac80211/Makefile
--- buildroot/package/kernel/mac80211/Makefile	2017-11-27 21:06:31.776053146 +0100
+++ buildroot/package/kernel/mac80211/Makefile	2017-11-29 15:18:15.189207621 +0100
@@ -10,11 +10,11 @@
 
 PKG_NAME:=mac80211
 
-PKG_VERSION:=2017-01-31
+PKG_VERSION:=2017-09-25
 PKG_RELEASE:=3
 PKG_SOURCE_URL:=http://mirror2.openwrt.org/sources
 PKG_BACKPORT_VERSION:=
-PKG_HASH:=75e6d39e34cf156212a2509172a4a62b673b69eb4a1d9aaa565f7fa719fa2317
+PKG_HASH:=1649144a6fd389ab8d58c207588c2a201dad22fe9acf86cb05e99ca8deb52e33
 
 PKG_SOURCE:=compat-wireless-$(PKG_VERSION)$(PKG_BACKPORT_VERSION).tar.xz
 PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/compat-wireless-$(PKG_VERSION)
